src/net_processing.cpp

```diff
@@ -3181,9 +3181,10 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, Peer& peer,
 void PeerManagerImpl::ProcessInvalidTx(NodeId nodeid, const CTransactionRef& ptx, const TxValidationState& state,
                                        bool maybe_add_extra_compact_tx)
 {
+    AssertLockHeld(m_tx_download_mutex);
     AssertLockNotHeld(m_peer_mutex);
     AssertLockHeld(g_msgproc_mutex);
-    AssertLockHeld(cs_main);
+    /* AssertLockHeld(cs_main); */
```

Comment:
This commented line maybe requires more explanation?

Reply:
I missed deleting that. Fixed, thanks!